TypeScript compilation errors

I'm getting:

```
node_modules/preact-mdl/dist/index.d.ts(12,41): error TS2314: Generic type 'ComponentProps<C>' requires 1 type argument(s).
node_modules/preact-mdl/dist/index.d.ts(13,34): error TS2314: Generic type 'ComponentProps<C>' requires 1 type argument(s).
node_modules/preact-mdl/dist/index.d.ts(15,31): error TS2314: Generic type 'ComponentProps<C>' requires 1 type argument(s).
```

This can be fixed by changing `preact.ComponentProps` to `preact.ComponentProps<any>` in the `.d.ts`.

1. Is this happening for everyone?
2. Should I just send a PR that fixes it with `<any>`?

same for me ... 

ERROR in /preact/preact-mdl-example/node_modules/preact-mdl/dist/index.d.ts
(12,41): error TS2314: Generic type 'ComponentProps<C>' requires 1 type argument(s).

ERROR in /preact/preact-mdl-example/node_modules/preact-mdl/dist/index.d.ts
(13,34): error TS2314: Generic type 'ComponentProps<C>' requires 1 type argument(s).

ERROR in /preact/preact-mdl-example/node_modules/preact-mdl/dist/index.d.ts
(15,31): error TS2314: Generic type 'ComponentProps<C>' requires 1 type argument(s).

when updating to 

    "preact": "^8.2.1",
    "preact-mdl": "^2.2.1",
    "preact-router": "^2.5.5",

with original files error due to missing FunctionalComponents .... so no luck at all...

--- a/docs.md
+++ b/docs.md
@@ -245,6 +245,7 @@ Snackbar
 
 **Parameters**
 
+-   `args` **...any** 
 -   `floating-label`  = false
 -   `multiline`  = false
 -   `expandable`  = false

--- a/docs.md
+++ b/docs.md
@@ -245,7 +245,7 @@ Snackbar
 
 **Parameters**
 
--   `args` **...any** 
+-   `args` **...this** 
 -   `floating-label`  = false
 -   `multiline`  = false
 -   `expandable`  = false

--- a/src/index.d.ts
+++ b/src/index.d.ts
@@ -9,10 +9,10 @@ export class MaterialComponent<PropsType, StateType> extends preact.Component<Pr
     ripple: boolean;
     upgradedBase: HTMLElement;
 
-    createMdlClasses(props: PropsType & preact.ComponentProps): string[];
-    mdlRender(props: PropsType & preact.ComponentProps): preact.VNode;
+    createMdlClasses(props: PropsType & preact.ComponentProps<any>): string[];
+    mdlRender(props: PropsType & preact.ComponentProps<any>): preact.VNode;
     preserveMdlDom(base: HTMLElement, r: HTMLElement): void;
-    render(props: PropsType & preact.ComponentProps, state: StateType): preact.VNode;
+    render(props: PropsType & preact.ComponentProps<any>, state: StateType): preact.VNode;
 }
 
 export interface HTMLProps extends JSX.HTMLAttributes { }

--- a/src/index.d.ts
+++ b/src/index.d.ts
@@ -9,10 +9,10 @@ export class MaterialComponent<PropsType, StateType> extends preact.Component<Pr
     ripple: boolean;
     upgradedBase: HTMLElement;
 
-    createMdlClasses(props: PropsType & preact.ComponentProps<any>): string[];
-    mdlRender(props: PropsType & preact.ComponentProps<any>): preact.VNode;
+    createMdlClasses(props: PropsType & preact.ComponentProps<this>): string[];
+    mdlRender(props: PropsType & preact.ComponentProps<this>): preact.VNode;
     preserveMdlDom(base: HTMLElement, r: HTMLElement): void;
-    render(props: PropsType & preact.ComponentProps<any>, state: StateType): preact.VNode;
+    render(props: PropsType & preact.ComponentProps<this>, state: StateType): preact.VNode;
 }
 
 export interface HTMLProps extends JSX.HTMLAttributes { }

